addons/mail/static/src/models/timer.js

```diff
@@ -0,0 +1,186 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
+import { makeDeferred } from '@mail/utils/deferred';
+import { TimerClearedError } from '@mail/utils/timer_cleared_error';
+
+registerModel({
+    name: 'Timer',
+    identifyingFields: [[
+        'otherMemberLongTypingInThreadTimerOwner',
+        'threadAsCurrentPartnerInactiveTypingInThreadTimerOwner',
+        'threadAsCurrentPartnerLongTypingInThreadTimerOwner',
+    ]],
+    lifecycleHooks: {
+        _willDelete() {
+            this.messaging.browser.clearTimeout(this.timeoutId);
+            if (!this.timeoutDeferred) {
+                return;
+            }
+            this.timeoutDeferred.reject(new TimerClearedError());
```

Comment:
Note: it was using `this.id` in previous code, but it was never introduced, so it was always `undefined`.
I just dropped this.